libs/wingc/src/type_check.rs

```diff
@@ -2790,6 +2805,8 @@ impl<'a> TypeChecker<'a> {
 			format!("\"{}\"", first_expected_type)
 		};
 
+		// check phase
```

Comment:
This seems like the right place to see if the difference between the two types is only the phase and augment the diagnostic accordingly.